core/templates/dev/head/exploration_editor/main_tab/StateEditor.js

```diff
@@ -77,35 +79,27 @@ oppia.controller('StateEditor', [
         $scope.isInteractionShown = true;
       }
 
+      $scope.$on('externalSave', function() {
+        if ($scope.contentEditorIsOpen) {
+          $scope.saveTextContent();
+        }
+      });
+
       $rootScope.loadingMessage = '';
     };
 
     $scope.openStateContentEditor = function() {
       if (editabilityService.isEditable()) {
-        $scope.contentMemento = angular.copy($scope.content);
+        $scope.contentEditorIsOpen = true;
       }
     };
 
     $scope.saveTextContent = function() {
```

Comment:
Ah I see--that's a good point. The overall lack of parity is a bit strange though:
- ``openStateContentEditor``
- ``saveTextContent``
- ``cancelEdit``

I suggest making these more consistent, but it's optional. I don't feel too strongly about it.